### ISSUE TITLE
preprocess not called again

### ISSUE THREAD

**Issue by andshape** [odysseyscience/react-s3-uploader#86]
I have an issue when the 'next' callback function is not called in the preprocess handler. The issue is that preprocess won't be called again if I select a new image. It might simply be that I am using this library wrong so any input on what I am doing wrong is greatly appreciated. The code to I have now is:

```
ReactS3Uploader
        signingUrl="/api/s3/sign"
        accept="image/*"
        preprocess={(file, next) => {
          console.log("S3 start upload", file);
          //next(file) //temp
        }}
        onProgress={p => console.log("S3 progress", p)}
        onError={e => console.log("S3 error", e)}
        onFinish={status => {
          console.log("S3 finished upload", status);
          }}
        contentDisposition="auto"
        uploadRequestHeaders={{}}
      />
```

Note that I have explicitly commented out the call to next. This is to simulate the situation where uploading the image is aborted for some reason. According to the docs this should be legal:

> The preprocess(file, next) prop provides an opportunity to do something before the file upload begins, modify the file (scaling the image for example), or **abort the upload by not calling next(file)**.

This successfully aborts the upload _but_ then I am not allowed to retry the situation (i.e. upload a new image). I _can_ select a new image, but preprocess is not called again for the (new) image.

**Comment by seanadkinson** [comment]
Hmm, I'm not seeing anything in the code that would keep the `preprocess` function from being called again.  According to [this line](https://github.com/odysseyscience/react-s3-uploader/blob/master/s3upload.js#L48), it would always be called.

Can you do any debugging or console logging to see what is going on better?  I'm swamped right now, so might not be able to look at this for a little bit.

Thanks for creating this issue though!

**Comment by simdax** [comment]
Yes, I can say the same thing for me. Not calling 'next' in the preprocess make this function never ever callable again :(
